ssaEmbeddedCore/ports/micropython/scripts/flash.sh

```
@@ -0,0 +1,380 @@
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Logging functions
+error() { echo -e "${RED}[ERROR]${NC} $1"; }
+success() { echo -e "${GREEN}[SUCCESS]${NC} $1"; }
+info() { echo -e "${BLUE}[INFO]${NC} $1"; }
+debug() { 
+    if [[ "$DEBUG" == "true" ]] then
```

Shellcheck (0.10.0)

[warning] Use semicolon or linefeed before 'then' (or quote to make it literal).

(SC1010)